src/Sarif.Viewer.VisualStudio.Core/CodeFlowToTreeConverter.cs

```diff
@@ -68,5 +69,44 @@ internal static List<AnalysisStepNode> Convert(CodeFlow codeFlow, Run run, int r
 
             return root.Children;
         }
+
+        internal static List<AnalysisStepNode> ToFlatList(CodeFlow codeFlow, Run run, int resultId, int runIndex)
+        {
+            var results = new List<AnalysisStepNode>();
+
+            ThreadFlow threadFlow = codeFlow.ThreadFlows?[0];
+
+            if (threadFlow != null)
+            {
+                // nestingLevel is an integer value which is greater than or equals to 0
+                // according to schema http://json.schemastore.org/sarif-2.1.0-rtm.5.
+                // min nesting level can be used to offset nesting level starts from value greater than 0.e
+                int minNestingLevel = threadFlow.Locations.Min(l => l.NestingLevel);
+
+                foreach (ThreadFlowLocation location in threadFlow.Locations)
+                {
+                    ArtifactLocation artifactLocation = location.Location?.PhysicalLocation?.ArtifactLocation;
+
+                    if (artifactLocation != null)
+                    {
+                        if (artifactLocation.Uri == null && artifactLocation.Index > -1)
```

Comment:
NIT: these could be collapsed, but I see the readability value of doing it this way.

Reply:
combined into one if condition